src/odemis/driver/test/xt_client_test.py

```diff
@@ -70,56 +70,75 @@ def setUp(self):
-        self.assertNotEqual(self.stage.position.value, pos)
+        self.assertNotEqual(self.stage.position.value, init_pos)
 
-        init_pos = self.stage.position.value.copy()  # returns [x, y, z, r, t]  in [m]
+        new_pos = self.stage.position.value.copy()  # returns [x, y, z, r, t]  in [m]
         # move stage to a different position
-        position = {'x': init_pos['x'] - 1e-6, 'y': init_pos['y'] - 2e-6, 'z': 10e-6}  # [m]
+        position = {'x': new_pos['x'] - 1e-6, 'y': new_pos['y'] - 2e-6, 'z': 10e-6}  # [m]
         f = self.stage.moveAbs(position)
         f.result()
-        self.assertNotEqual(self.stage.position.value, pos)
+        self.assertNotEqual(self.stage.position.value, new_pos)
 
         stage_position = self.stage.position.value.copy()
         # test move_stage method actually moves HW by the requested value
         self.assertAlmostEqual(stage_position['x'], position['x'])
         self.assertAlmostEqual(stage_position['y'], position['y'])
         self.assertAlmostEqual(stage_position['z'], position['z'])
         # test relative movement
-        init_pos = self.stage.position.value.copy()  # returns [x, y, z, r, t]  in [m]
+        new_pos = self.stage.position.value.copy()  # returns [x, y, z, r, t]  in [m]
         relative_position = {'x': 100e-6, 'y': 200e-6}  # [m]
         f = self.stage.moveRel(relative_position)
         f.result()
 
         stage_position = self.stage.position.value.copy()
         # test move_stage method actually moves HW by the requested value
-        self.assertAlmostEqual(stage_position['x'], relative_position['x'] + init_pos['x'])
-        self.assertAlmostEqual(stage_position['y'], relative_position['y'] + init_pos['y'])
+        self.assertAlmostEqual(stage_position['x'], relative_position['x'] + new_pos['x'])
+        self.assertAlmostEqual(stage_position['y'], relative_position['y'] + new_pos['y'])
 
         # move to a position out of range -> should be impossible
         position = {'x': 1, 'y': 300}  # [m]
         with self.assertRaises(Exception):
             f = self.stage.moveAbs(position)
             f.result()
+        # Move stage back to initial position
+        self.stage.moveAbs(init_pos)
+
+    def test_stop_stage_movement(self):
+        """Test that stopping stage movement, stops moving the stage."""
+        if self.xt_type == 'xttoolkit':
+            self.skipTest("Skip because the microscope stage is not used, an external stage is used.")
+        init_pos = self.microscope.get_stage_position()  # returns [x, y, z, r, t]  in [m]
+        # move stage to a different position
+        position = {'x': init_pos['x'] - 1e-6, 'y': init_pos['y'] - 2e-6, 'z': 10e-6}  # [m]
+        self.microscope.move_stage(position)
+        self.microscope.stop_stage_movement()
+        time.sleep(1)
+        self.assertFalse(self.microscope.stage_is_moving())
+        # Move stage back to initial position
+        self.microscope.move_stage(init_pos)
```

Comment:
I'm not a fan of using these internal functions. The testing should only test the functions provided by the Odemis API. Here that'd be `moveRel()`, `moveAbs()`, `future.cancel()`, `.stop()`, and `.position`.

If you really want to test these internal functions, maybe make it really clear, by moving this test to a separate TestCase class, or appending to the function name `_internal`.
There are several other testing functions in the same vein that could be adjusted the same way.